src/univariate/continuous/inversegaussian.jl

```diff
@@ -186,10 +186,14 @@ function suffstats(::Type{InverseGaussian}, x::AbstractVector{<:Real}, w::Abstra
     if length(w) != n
         throw(DimensionMismatch("Inconsistent argument dimensions."))
     end
-
-    sx = dot(w, x)
-    sinvx = sum(w[i]/x[i] for i in eachindex(x))
-    sw = sum(w)
+    sx = zero(eltype(x))
+    sinvx = zero(eltype(x))
+    sw = zero(eltype(x))
```

Comment:
sorry to be a pain but I think this is potentially problematic in pathological cases where for instance `x` is given in floating point and `w` in Integer which is allowed by the function signature afaik. I think what I had written may be preferable i.e. (using your notations this time):

```
sw = zero(eltype(w))
sx = zero(one(eltype(w)) * one(eltype(x)))
sinvx = zero(one(eltype(w))/one(eltype(x)))
```

Reply:
Makes sense. Though better use `zero` directly instead of `zero(one(...))`.

Reply:
Just writing a generator would have the advantage of being easier maintenance.

Reply:
Yeah, it depends on whether performance really matters in that particular piece of code.

Reply:
agreed on principle with @mschauer, but yeah this code might be used with lots of data points and the difference is significant (see benchmarks).
`promote_type` will help make it cleaner

Reply:
and no problem @tlienart, in that case, pain is the path to correctness